benchmarks/test_sharded_parity.py

```diff
@@ -144,27 +142,15 @@ def plugin_parity_test(
     ddp_model = model_cls()
     use_cuda = gpus > 0
 
-    trainer = Trainer(
-        fast_dev_run=True,
-        max_epochs=1,
-        gpus=gpus,
-        precision=precision,
-        accelerator='ddp_spawn',
-    )
+    trainer = Trainer(fast_dev_run=True, max_epochs=1, gpus=gpus, precision=precision, accelerator="ddp_spawn")
```

Comment:
I would prefer to keep this as multiline :confused:

Reply:
> I would prefer to keep this as multiline 😕

why? It's one line because it fits in the 120 max length.

So it's consistent.

> would need the magic trailing comma right?

Yes but we are using `--skip-magic-trailing-comma` so since it can be in a single line, it would get removed.

Reply:
> why? It's one line because it fits in the 120 max length.

yes, but is about readability, have the trainer formatted always the same way, so for example as multiline so you just simply look and see, also any argument change is much easier to recognise, then mentally parsing full line :D

Reply:
I vote for having the magic comma with this behavior: 
https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma

Reply:
The issue with the magic comma is that you have to be mindful about where you add it, as the formatted does not know whether you manually added it because you want a specific line to be expanded or because it was there accidentally.

For example, take this input snippet with different alternatives:

```python
if isinstance(input, (list, tuple)):
    pass

if isinstance(input, (list, tuple,),):
    pass

if isinstance(
    input,
    (
        list,
        tuple,
    ),
):
    pass

if isinstance(
    input,
    (
        list,
        tuple
    )
):
    pass
```

`black --skip-magic-trailing-comma file.py`

```python
if isinstance(input, (list, tuple)):
    pass

if isinstance(input, (list, tuple)):
    pass

if isinstance(input, (list, tuple)):
    pass

if isinstance(input, (list, tuple)):
    pass
```

`black file.py`

```python
if isinstance(input, (list, tuple)):
    pass

if isinstance(
    input,
    (
        list,
        tuple,
    ),
):
    pass

if isinstance(
    input,
    (
        list,
        tuple,
    ),
):
    pass

if isinstance(input, (list, tuple)):
    pass
```

Considering the whole expression can be fit into one line (because it's length is <120), I personally don't like that it stays expanded because a trailing comma was there.

Reply:
> have the trainer formatted always the same way, so for example as multiline so you just simply look and see, also any argument change is much easier to recognise, then mentally parsing full line :D

The formatter checks if the line can fit under the line length limit, and formats it that way.

I would find it weird (and inconsistent) that we would break that rule one class in particular (`Trainer`)

Reply:
Definitely a hard problem :) 

I definitely prefer `black --skip-magic-trailing-comma file.py` in the above context but agree with Jirka on readability of expending Trainer arguments onto multiple lines in the tests.

I guess we can't have everything ;)

Reply:
we can fake it by repeating default args to have the line sufficiently long to get it split to multiple :D